test/conversion_test.exs

```diff
@@ -18,15 +18,15 @@ defmodule ExDimension.ConversionTest do
 
   test "convert complex units to another" do
     sq_in = ExDimensions.Area.square_inches(4)
-    mm_sq = sq_in ~> (ExDimensions.Spatial.Millimeters ^^^ 2)
+    mm_sq = sq_in ~> dimensions(ExDimensions.Spatial.Millimeters, 2)
```

Comment:
I think there are 2 choices here - 

1.  Maintain the operator DSL but use a different one.  There aren't many great ones left that look like exponentiation.  Some options that don't look utterly horrible are `<>`, `|>`, or maybe `::`.
2. Stick with a function but make it clear it is exponentiating the units.  So something like `exp` or `expon` or `pow`.

Reply:
To be explicit, the situation I'm trying to address is:
- we build FROS with `mix compile --warnings-as-errors` which I'd like to preserve
- When building with Elixir 1.12+ any usage of `^^^` operator as a specific name is a non-configurable warning, and thus will fail our builds due to above.

We must not override any syntax that has significant meaning in vanilla Elixir code, so `|>` and `<>` are out.

I don't want to explore conditionally compiling with/without the infix operator depending on what Elixir version you use because it's too fragile.

Custom infix operators in Elixir are actually quite uncommon IME and I personally find them opaque to read despite some previous Haskell experience where it's quite popular.

Reply:
Given the options my vote is for `pow`.

Reply:
`pow` isn't bad.  The only problem with `pow` is that it usually means just raising a simple value to a power.  Given the context and the arguments it would be pretty clear that that's not exactly what's going on here.  It would be nice if the function name made that distinction clearer, but I'm not sure what that would be.  As one who was unfamiliar with this module before viewing this PR, I find `pow` to be more readable than `^^^`.

Reply:
Renamed in ae1fa8f6f199cb905cff14a8076ab3353a397735.